ModuleInstall/PackageManager/PackageManager.php

```diff
@@ -438,29 +439,65 @@ public function validate_manifest($manifest)
             $matches_empty = true;
             if (isset($manifest['acceptable_sugar_versions']['exact_matches'])) {
                 $matches_empty = false;
                 foreach ($manifest['acceptable_sugar_versions']['exact_matches'] as $match) {
                     if ($match == $sugar_version) {
-                        $version_ok = true;
+                        $version_sugar_ok = true;
+                        $GLOBALS['log']->debug("Passed AcceptableSugarVersions");
+                        break;
                     }
                 }
             }
-            if (!$version_ok && isset($manifest['acceptable_sugar_versions']['regex_matches'])) {
+            if (!$version_sugar_ok && isset($manifest['acceptable_sugar_versions']['regex_matches'])) {
                 $matches_empty = false;
                 foreach ($manifest['acceptable_sugar_versions']['regex_matches'] as $match) {
                     if (preg_match("/$match/", $sugar_version)) {
-                        $version_ok = true;
+                        $version_sugar_ok = true;
+                        $GLOBALS['log']->debug("Passed AcceptableSugarVersions");
+                        break;
                     }
                 }
             }
 
-            if (!$matches_empty && !$version_ok) {
+            if (!$matches_empty && !$version_sugar_ok) {
+                $GLOBALS['log']->debug("Error with AcceptableSugarVersions");
                 die($mod_strings['ERROR_VERSION_INCOMPATIBLE'] . $sugar_version);
             }
         }
+        if (isset($manifest['acceptable_suitecrm_versions'])) {
+            $GLOBALS['log']->debug("Getting AcceptableSuiteCRMVersions");
+            $version_suitecrm_ok = false;
+            $matches_empty = true;
+            if (isset($manifest['acceptable_suitecrm_versions']['exact_matches'])) {
+                $matches_empty = false;
+                foreach ($manifest['acceptable_suitecrm_versions']['exact_matches'] as $match) {
+                    if ($match == $suitecrm_version) {
+                        $version_suitecrm_ok = true;
+                        $GLOBALS['log']->debug("Passed AcceptableSuitecrmVersions");
+                        break;
+                    }
+                }
+            }
+            if (!$version_suitecrm_ok && isset($manifest['acceptable_suitecrm_versions']['regex_matches'])) {
+                $matches_empty = false;
+                foreach ($manifest['acceptable_suitecrm_versions']['regex_matches'] as $match) {
+                    if (preg_match("/$match/", $suitecrm_version)) {
+                        $version_suitecrm_ok = true;
+                        $GLOBALS['log']->debug("Passed AcceptableSuitecrmVersions");
+                        break;
+                    }
+                }
+            }
+
+            if (!$matches_empty && !$version_suitecrm_ok) {
+                $GLOBALS['log']->debug("Error with AcceptableSuiteCRMVersions");
```

Comment:
this error message is exact duplication of this one above: https://github.com/salesagility/SuiteCRM/pull/6479/files#diff-3c30fdcb40dd2a7e029b553efad3ba76R467
It doesn't really make sense what is the error looking die() message on screen or in the log file.
Please add more informative message to see if `acceptable_sugar_versions` or  `acceptable_suitecrm_versions` was wrong in the manifest.

Reply:
@gymad don't understand your message. We are using the same process and a specific string saying that suitecrm version declared in manifest is not compatible with current/actual suitecrmversion (like the previous code block is doing the same on sugar version !)

There's nothing more here instead of an admin to check the manifest file and the suitecrm version.

My idea was not to recode from scratch the validate_manifest method !